### ISSUE TITLE
training problem

### ISSUE THREAD

**Issue by l772308126** [NVlabs/EdgeRunner#7]
Sorry to bother you again. It's really a nice work! I have found two problem when I was training the main.py.
1. ![image](https://github.com/user-attachments/assets/4aa8720c-a875-4c62-8d0f-559521abef87)
I think this code is wrong，because the unpad_input function has five outputs。
![image](https://github.com/user-attachments/assets/c70d6c36-d1aa-4080-affa-720678ba4133)
So，it will report an error when I run the training code of main.py. I guess it should be revised to 'q, indices_q, cu_seqlens_q, max_len_q, _ = unpad_input(q, mask_q)' and 'k, indices_kv, cu_seqlens_kv, max_len_kv, _ = unpad_input(k, mask_kv)'. Is that correct?

2. I found that the tokenized sequences obtained by mesh through tokenizer always have vertices 3 and 4. I counted the number of vertices 3 and 4. It even exceeds the sum of the occurrences of the other top 10 vertices. As a result, the final predicted results are all 3 and 4. Have you ever encountered such a problem during training? How to solve it?
![image](https://github.com/user-attachments/assets/c72ebdf6-1a24-4bef-9e86-8b6861f2cc0e)

**Comment by l772308126** [comment]
I use the g-objaverse dataset and randomly select some glb files containing faces less than 4000.

**Comment by creiser** [comment]
Regarding 1), yes, your fix is correct.

**Comment by ashawkey** [comment]
Thanks for reporting this! 
It seems flash-attn's latest version changes some API, I will fix this later.
Token 3 and 4 are the next/prev token (or left/right) as defined in the edgebreaker traversal. They do appear more frequently (before every new triangle), but after convergence they should be generated correctly.

**Comment by l772308126** [comment]
> Thanks for reporting this! It seems flash-attn's latest version changes some API, I will fix this later. Token 3 and 4 are the next/prev token (or left/right) as defined in the edgebreaker traversal. They do appear more frequently (before every new triangle), but after convergence they should be generated correctly.

Thanks for your reply! When can we get the fixed version? I want to use your code as my baseline. At the same time, if I change the version of the flash-attn, which version is correct?

**Comment by ashawkey** [comment]
Hi, it seems flash-attn >= 2.7.0 changes the API. The code should be fixed now, welcome to try it out.

**Comment by l772308126** [comment]
> Hi, it seems flash-attn >= 2.7.0 changes the API. The code should be fixed now, welcome to try it out.

Hi, I changed the version of flash-attn to 2.6.3 and downloaded the fixed code at the same time. It also has the same problem, the final predicted results are all 3 and 4. I use torch2.4.1+cuda12.1. The other environment settings are following your commands.

**Comment by ashawkey** [comment]
Hi, could you provide more details? e.g., datasets, training resources and iterations?

**Comment by l772308126** [comment]
> Hi, could you provide more details? e.g., datasets, training resources and iterations?

I’m sorry to keep asking. 

I used the g-objaverse datasets and randomly selected 60 samples from them. 2 A6000 are used to do multi-GPU training. I just modified the file path in the provider.py. Nothing has been changed except for batchsize, learning rate and max_face_length. All environment configurations are configured according to requirements.lock. The version of python is 3.10 and pytorch is torch2.4.1+cuda12.1. I used this config to train the ArAE:
        point_encoder_mode='embed',
        kl_weight=1e-8,
        discrete_bins=512,
        use_num_face_cond=True,
        use_decimate_aug=True,
        cond_mode='point',
        num_cond_tokens=2049,
        freeze_encoder=False,
        use_meto=True,
        meto_backend='LR_ABSCO',
        max_face_length=8000,
        max_seq_length=40960,
        align_posemb='right',
        batch_size=2,
        hidden_dim=1536,
        num_heads=16,
        num_layers=24,
        gradient_accumulation_steps=1,
        lr=5e-5,
        warmup_ratio=0,
        num_epochs=100,
        eval_mode='loss',
        workspace='.../'.
